pennylane/__init__.py

```diff
@@ -103,7 +103,6 @@
 
 # collections needs to be imported after all other pennylane imports
 from .collections import QNodeCollection, dot, map, sum
-from pennylane.pauli import grouping  # pylint:disable=wrong-import-order
```

Comment:
Should we keep this in case there's some external code that uses `qml.grouping.`?

It would also be nice that a deprecation warning is raised every time a user calls `qml.grouping`.

Reply:
A deprecation warning is raised, and now the pauli module is imported as `from pennylane import pauli`. Any queries to `qml.grouping` now raise a deprecation warning and re-route the request to `qml.pauli`